src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder resultStr = new StringBuilder();
+        resultStr.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            resultStr.append(System.lineSeparator()).append(name).append(" - ");
+            int sum = 0;
+            for (String d : data) {
+                String[] splitData = d.split(" ");
+                if (splitData[1].equals(name) && isRelevantDate(splitData[0], dateFrom, dateTo)) {
+                    sum += Integer.parseInt(splitData[2]) * Integer.parseInt(splitData[3]);
+                }
+            }
+            resultStr.append(sum);
+        }
+        return resultStr.toString();
+    }
+
+    private boolean isRelevantDate(String dateStr, String dateFromStr, String dateToStr) {
+        LocalDate date = LocalDate.parse(dateStr, DATE_TIME_FORMATTER);
+        LocalDate dateFrom = LocalDate.parse(dateFromStr, DATE_TIME_FORMATTER);
+        LocalDate dateTo = LocalDate.parse(dateToStr, DATE_TIME_FORMATTER);
```

Comment:
`isRelevantDate` name is quite confusing, don't you think? relevant to what? which criteria?

therefore I would name this method accordingly to what it does - `isDateBetween`

also I suggest renaming the parameters to `(String target, String from, String to)` - more readable

now let's extract the parsing to a separate method to avoid violation of [DRY principle](https://uk.wikipedia.org/wiki/Don%27t_repeat_yourself):
```
private LocalDate parse(String date) {
  return LocalDate.parse(date, DATE_TIME_FORMATTER);
}
```

```suggestion
    private boolean isDateBetween(String target, String from, String to) {
        LocalDate date = parse(target);
        LocalDate dateFrom = parse(from);
        LocalDate dateTo = parse(to);
```

Reply:
you haven't addressed this remarks completely :)